lib/pc_schema.c

```diff
@@ -635,3 +635,36 @@ pc_schema_get_size(const PCSCHEMA *s)
 {
 	return s->size;
 }
+
+/**
+* Return true if s1 and s2 have the same dimensions at the same
+* positions, otherwise return false.
+*/
+uint32_t
+pc_schema_similar(const PCSCHEMA *s1, const PCSCHEMA *s2)
+{
+	size_t i;
+
+	if ( s1->ndims != s2->ndims )
+		return PC_FALSE;
+
+	for ( i = 0; i < s1->ndims; i++ )
+	{
+		PCDIMENSION *s1dim = s1->dims[i];
+		PCDIMENSION *s2dim = s2->dims[i];
+
+		if ( strcmp(s1dim->name, s2dim->name) != 0 )
+			return PC_FALSE;
+
+		if ( s1dim->interpretation != s2dim->interpretation )
+			return PC_FALSE;
+
+		if ( s1dim->scale != s2dim->scale )
+			return PC_FALSE;
+
+		if ( s1dim->offset != s2dim->offset )
+			return PC_FALSE;
```

Comment:
Let me think... My concern is not to say that all should be in this PR (which is already really good), but I would like to avoid a situation where we decide on an API now that would make it difficult later to implement that without breaking too much the SQL API.

Let me try to sum up :

I see 3 possible boolean options : 
- whether you want to fail on the slow path to ensure a fast path (now a fast path is defined as requiring only a memcpy or recompressing)
- whether you want to match dimensions by position or name (now set to "byName")
- whether you want to want to change the raw values if scale/offset is modified (now set to "rescale" for each dimension)

The last 2 have clear semantics, the first one a little less since you uncovered that differing compressions might trigger a no so fast recompression path, but that is ok. I can argue however that the "rescale" option is rather in the scope of a PC_transform function (when used to change between 2 pcids that have the same srid)

I see 2 ways to provide default values when required by missing dimensions : 
- using a default value common to all dimensions (current implementation)
- using a pcpoint (which would provide both the default value of each dimension and the new pcid)
(an array of default values, 1 for each dimension, is redundant with the pcpoint option, as you can build a pcpoint from it)

If the following 3 properties are true, a fast path may then be used (memcpy or recompression only) :
- The schemas have the same number of dimensions with matching interpretations
- There is no need to reorder dimensions (i.e. : schemas are matched by position, or by name with no need to reshuffle the dimensions)
- There is no need to change the scale/offset of the raw values (i.e. scale/offsets are matching, or the user wants to change the scale/offset without modifying the raw values)